libr/anal/type.c

```diff
@@ -330,35 +279,81 @@ static RAnalBaseType *get_union_type(RAnal *anal, const char *sname) {
 	return NULL;
 }
 
+static RAnalBaseType *get_typedef_type(RAnal *anal, const char *sname) {
+	r_return_val_if_fail (anal && sname, NULL);
+
+	RAnalBaseType *base_type = r_anal_base_type_new (R_ANAL_BASE_TYPE_KIND_TYPEDEF);
+	if (!base_type) {
+		return NULL;
+	}
+
+	base_type->type = get_type_data (anal->sdb_types, "typedef", sname);
+	if (!base_type->type) {
+		goto error;
+	}
+	return base_type;
+
+error:
+	r_anal_base_type_free (base_type);
+	return NULL;
+}
+
+static RAnalBaseType *get_atomic_type(RAnal *anal, const char *sname) {
+	r_return_val_if_fail (anal && sname, NULL);
+
+	RAnalBaseType *base_type = r_anal_base_type_new (R_ANAL_BASE_TYPE_KIND_ATOMIC);
+	if (!base_type) {
+		return NULL;
+	}
+
+	base_type->type = get_type_data (anal->sdb_types, "type", sname);
+	if (!base_type->type) {
+		goto error;
+	}
+
+	RStrBuf key;
+	r_strbuf_init (&key);
+	r_strbuf_setf (&key, "type.%s.size", sname);
+	base_type->size = sdb_num_get (anal->sdb_types, r_strbuf_get (&key), 0);
```

Comment:
```suggestion
	base_type->size = sdb_num_get (anal->sdb_types, sdb_fmt ("type.%s.size", sname), 0);
```

Reply:
`sdb_fmt()` is bad

Reply:
Yeah we know, but it is used everywhere anyway. It won't be this single place to save r2world :D (nor to ruin it btw)

Reply:
Well we want to get rid of it eventually, so better not use it at all in new code.

Reply:
I know. But we had 5 different implementations of it and end up to the trash

Reply:
im totally fine to replace sdb_fmt with strbuf, so for me this thread is resollved